compiler/src/iree/compiler/Dialect/LinalgExt/IR/LinalgExtOps.cpp

```diff
@@ -1576,14 +1626,29 @@ LogicalResult Im2colOp::verify() {
     return op->emitOpError("expected one output operand");
   }
 
-  // TODO(Max191): Support cases with more than 1 m or k dimension, and remove
-  // the check for a single m_offset and k_offset.
-  if (getMixedMOffset().size() != 1) {
-    return op->emitOpError("expected one m_offset");
+  // Verify offsets and strides
+  SmallVector<OpFoldResult> kOffset = getMixedKOffset();
+  SmallVector<OpFoldResult> mOffset = getMixedMOffset();
+  SmallVector<OpFoldResult> kStrides = getMixedKStrides();
+  SmallVector<OpFoldResult> mStrides = getMixedMStrides();
+  if (kOffset.size() != kStrides.size()) {
+    return op->emitOpError("expected the same size k_offset and k_strides");
+  }
+  if (mOffset.size() != mStrides.size()) {
+    return op->emitOpError("expected the same size m_offset and m_strides");
   }
-  if (getMixedKOffset().size() != 1) {
-    return op->emitOpError("expected one k_offset");
+  std::optional<int64_t> constInnerKStrides =
+      getConstantIntValue(kStrides.back());
```

Comment:
I think we should check if the size of k/mStrides is at least one or not. Otherwise, it could crash in the op verifier.